wrapper.py

```diff
@@ -331,3 +330,15 @@ def load_h5(filename, path='', lazy=False):
                   "will be removed in the next release. Please use load() instead.",
                   DeprecationWarning)
     return load(filename, path=path, lazy=lazy)
+
+
+# Helper function
+def _accumulate(iterator):
+    """
+    Create a generator to iterate over the accumulated
+    values of the given iterator.
+    """
+    total = 0
+    for item in iterator:
+        yield total
```

Comment:
could you even simplify this further, by returning `total, item`? the loop would then just read:
```python
for counter, l in _accumulate(f.attrs['oldshape']):
```